libs/clients/rsk/company-registry/src/lib/apiConfiguration.ts

```diff
@@ -0,0 +1,54 @@
+import { createEnhancedFetch, Request } from '@island.is/clients/middlewares'
+import { ConfigType, XRoadConfig } from '@island.is/nest/config'
+
+import { Configuration } from './gen/fetch'
+import { CompanyRegistryConfig } from './company-registry.config'
+import { caching } from 'cache-manager'
+import redisStore from 'cache-manager-ioredis'
+import { createRedisCluster } from '@island.is/cache'
+
+const registryEndpoint = /\d{10}$/
+
+export const ApiConfiguration = {
+  provide: 'CompanyRegistryClientApiConfiguration',
+  useFactory: (
+    config: ConfigType<typeof CompanyRegistryConfig>,
+    xroadConfig: ConfigType<typeof XRoadConfig>,
+  ) => {
+    const cache =
+      config.redis.nodes.length === 0
+        ? undefined
+        : {
+            cacheManager: caching({
+              store: redisStore,
+              ttl: 0,
+              redisInstance: createRedisCluster({
+                name: 'clients-company-registry',
+                nodes: config.redis.nodes,
+                ssl: config.redis.ssl,
+                noPrefix: true,
+              }),
+            }),
+            overrideCacheControl: (request: Request) => {
+              // Only cache company lookups. Not search.
+              if (request.url.match(registryEndpoint)) {
```

Comment:
FYI this could match the search path if the search term is the national Id of the company ;)

Reply:
Good point. I think that's fine. It's ok to cache search results as well. The goal here is to avoid filling redis up with search phrases (one letter at a time).